sdk/clientcore/core/src/main/java/io/clientcore/core/utils/configuration/Configuration.java

```diff
@@ -110,12 +118,8 @@ public final class Configuration {
      *
      * @param sources The configuration sources to use.
      */
-    public Configuration(ConfigurationSource... sources) {
-        if (sources == null || sources.length == 0) {
-            this.sources = null;
-        } else {
-            this.sources = Arrays.asList(Arrays.copyOf(sources, sources.length));
-        }
+    public static Configuration from(ConfigurationSource... sources) {
+        return new Configuration(sources);
```

Comment:
One thing we discussed yesterday, this should either throw on or remove null sources when the array is non-null and non-empty.